### DISCUSSION TITLE
BUG: flood error and then stuck

### DISCUSSION THREAD

**Discussion by johnbubak** [aahnik/tgcf#76] [Q&A] [answered]
Hi,
I'm runing 20 instances of tgcf script.
With 2 APi credentials.

Sometimes I get 
`INFO:telethon.client.users:Sleeping for 11s (0:00:11) on GetFullChannelRequest flood wait`

Next message is then send a bit later, what would be ok.
But, now it stuck.

How I can avoid this at all.

(I've read something about asynchroniuos and to avoid using sleep() in it)
+ I think one listener client per API Key would be optimal to avoid floods from script side.

Thanks for help

**Comment by aahnik** [comment, marked as answer]
why are you running 20 instances ? 

i hope the 20 instances are with different bots/users ? or are you using same user account ?

i cant understand why you need 20 instances ? 

do you need to forward from multiple sources to multiple destinations ?

then you may define that in the `tgcf.config.yml` . read this https://github.com/aahnik/tgcf/wiki/How-to-configure-tgcf-%3F

and also this https://github.com/aahnik/tgcf/discussions/68#discussioncomment-647742

**Reply by johnbubak** [reply]
> why are you running 20 instances ?
I'm connecting many to any as you thought.
With same credentials.

Thanks for advise. Think this is it. I've missunderstood how to configure.

Best
Jan
